admin/services/data-access/azure-table.data.service.js

```diff
@@ -1,36 +1,41 @@
 'use strict'
 
-const azure = require('azure-storage')
-const bluebird = require('bluebird')
-let azureTableService
+const { TableClient } = require('@azure/data-tables')
+const config = require('../../config')
+
+const connectionString = config.AZURE_STORAGE_CONNECTION_STRING
 
-/**
- * Promisify and cache the azureTableService library as it still lacks Promise support
- */
 const service = {
-  getPromisifiedAzureTableService: function getPromisifiedAzureTableService () {
-    if (azureTableService) {
-      return azureTableService
-    }
-    azureTableService = azure.createTableService()
-    bluebird.promisifyAll(azureTableService, {
-      promisifier: (originalFunction) => function (...args) {
-        return new Promise((resolve, reject) => {
-          try {
-            originalFunction.call(this, ...args, (error, result, response) => {
-              if (error) {
-                return reject(error)
-              }
-              resolve({ result, response })
-            })
-          } catch (error) {
-            reject(error)
-          }
-        })
+  retrieveEntity: async function retrieveEntity (tableName, partitionKey, rowKey) {
+    const client = TableClient.fromConnectionString(connectionString, tableName)
+    try {
+      const entity = await client.getEntity(partitionKey, rowKey)
+      return entity
+    } catch (error) {
+      if (error.details.odataError.code === 'ResourceNotFound') {
+        throw new Error(`entity not found with PartitionKey:${partitionKey} rowKey:${rowKey}`)
+      } else {
+        throw error
       }
-    })
+    }
+  },
 
-    return azureTableService
+  clearTable: async function clearTable (tableName) {
+    const tableClient = TableClient.fromConnectionString(connectionString, tableName)
+    const entityIterator = tableClient.listEntities()
+    const deletions = []
+    for await (const entity of entityIterator) {
```

Comment:
This could presumably take a really long time to complete if the table was really big.   I presume this is only for non-prod workloads.  If so, how can we surface that?